v2/lints/cabf_br/lint_ocsp_id_pkix_ocsp_nocheck_ext_not_included_test.go

```diff
@@ -0,0 +1,73 @@
+package cabf_br
+
+/*
+ * ZLint Copyright 2020 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import (
+	"testing"
+
+	"github.com/zmap/zlint/v2/lint"
+	"github.com/zmap/zlint/v2/test"
+)
+
+type TestCase struct {
+	Name           string
+	Filename       string
+	ExpectedResult lint.LintStatus
+}
+
+func RunTest(lintName string, testCases []TestCase, t *testing.T) {
+	t.Helper()
+	for _, tc := range testCases {
+		t.Run(tc.Name, func(t *testing.T) {
+			result := test.TestLint(lintName, tc.Filename)
+			if result.Status != tc.ExpectedResult {
+				t.Errorf("expected result %v was %v", tc.ExpectedResult, result.Status)
+			}
+		})
+	}
+}
+
+func TestOCSPIDPKIXOCSPNocheckExtNotIncluded(t *testing.T) {
+	testCases := []TestCase{
```

Comment:
It took me a while to figure out what the properties are that you're testing, since you're using "real" certs that have a number of fields.

Can you confirm my understanding:
* S/MIME CA Wrong
  * Includes OCSPSigning EKU, but lacks nocheck
  * `CheckApplies` is true (because EKU)
  * Result is a warning (because not clear it's in scope; in practice, it is, because the issuer is in-scope)
* S/MIME CA Correct
  * Omits OCSPSigning EKU entirely
  * `CheckApplies` is false (because EKU)
* TLS CA Wrong
  * Includes OCSPSigning EKU, omits nocheck
  * `CheckApplies` is true (because EKU)
  * Result is an error (because TLS serverAuth EKU is present)
* TLS CA Correct
  * Includes TLS serverAuth EKU, omits OCSP EKU
  * `CheckApplies` is false for this lint (because no OCSP EKU)
* Delegated Responder
  * Only includes OCSP Signing EKU, has nocheck
  * `CheckApplies` is true (because EKU)
  * Result is pass

If we look at how the code is structured, to obtain 100% coverage we need tests to ensure:
* OCSP EKU absent, (NA)
* OCSP EKU present, nocheck present (Pass)
* OCSP EKU present, nocheck absent, TLS EKU present (Error)
* OCSP EKU present, nocheck absent (Warning)

You have five tests, and it seems the "TLS CA Correct" and "SMIME CA Correct" are just testing the same thing. Am I overlooking something?

Reply:
I have created test certificates for all 32 possible combinations and updated the tests respectively